source/conversion.c

```diff
@@ -0,0 +1,129 @@
+#include "conversion.h"
+#include "draw.h"
+
+#include <png.h>
+
+static void rotate_agbr_counterclockwise(char** bufp, size_t size, size_t width)
+{
+    uint32_t* buf = (uint32_t*)*bufp;
+    uint32_t* out = malloc(size);
+
+    size_t pixel_count = (size/4);
+    size_t height = pixel_count/width;
+
+    for (uint32_t h = 0; h < height; ++h)
+    {
+        for (uint32_t w = 0; w < width; ++w)
+        {
+            size_t buf_index = (w + (h * width));
+            size_t out_index = (height * (width-1)) - (height * w) + h;
+
+            out[out_index] = buf[buf_index];
+        }
+    }
+
+    free(*bufp);
+    *bufp = (char*)out;
+}
+
+size_t bin_to_abgr(char** bufp, size_t size)
+{
+    size_t out_size = (size / 3) * 4;
```

Comment:
Been thinking about this, it may be cheaper to do `size * (4 / 3)` since the compiler can optimize the division out.

Reply:
No it can't, (4 / 3) == 1. You have to do x / 3 * 4 or x * 4 / 3

Reply:
You are correct, one of the constants would need to be turned into a float.

I wonder how float calculations compare to a pure integer division.